pandas/tests/plotting/test_groupby.py

```diff
@@ -65,3 +66,18 @@ def test_plot_kwargs(self):
 
         res = df.groupby("z").plot.scatter(x="x", y="y")
         assert len(res["a"].collections) == 1
+
+
+@td.skip_if_no_mpl
+@pytest.mark.parametrize("column", [None, "b"])
+@pytest.mark.parametrize("label", [None, "d"])
+def test_hist_with_legend(column, label):
+    index = Index(15 * [1] + 15 * [2], name="c")
+    df = DataFrame(np.random.randn(30, 2), index=index, columns=["a", "b"])
+    g = df.groupby("c")
+
+    g.hist(column=column, label=label, legend=True)
+    tm.close()
+    if column != "b":
+        g["a"].hist(label=label, legend=True)
+        tm.close()
```

Comment:
maybe to have ` _check_plot_works(df.hist, **your_assigned_kwargs)`? 

and also would be nice to test the if the labels in the legend are the group keys, e.g. using `self._check_legend_labels` method.

Reply:
_check_plot_works seems to not play well with DataFrameGroupby and SeriesGroupby, but I added the _check_legend_labels.